### PR TITLE
Multiple enhancement for Layouts and their Controls

### PR DESCRIPTION
**Pull request by linusha**
Once again, more changes than necessarily should be in one PR ended up in here, sorry.

The central theme of this PR is enhancements for layouts, and especially the layout controls in the properties panel.

- We now only show the Constraints controls when they are applicable.
- The resizing controls are now integrated within the shape controls. This is akin to how figma handles this.
- When the position/extent of a morph is managed by a layout, the appropriate input fields in the shape control are disabled.
- The "center" option for the constraints is now available (thanks to @SilvanVerhoeven). This also lead to a name change of the `ProportionalLayout`, which has become the  `ConstraintLayout`.
- The available constraint controls/resizing modes are updated on the fly, e.g. directly when adding a `TilingLayout` with the properties panel.

Here is a video, showcasing the new functionality:

https://user-images.githubusercontent.com/14252419/204611612-9f134c7c-db37-42fc-9e33-568d4ba213e1.mp4

As can be seen in the video, the tooling enforces some settings, namely:

- filling/hugging on layouts/items in layouts does disable `wrapSubmorphs` on that layout
-  when a container only contains fill items, it cannot be set to hugging

This should be consistent with the enforcement of the logic in the code. However, this begs the question of how affordable such a UI is, e.g., because the disabled elements at the moment do not come with any kind of explanation.

---

# To Do

- [x] investigate effects on Rick's Dashboard

### CODE REVIEW

File: lively.ide/studio/styling.js [hunk 1]
```diff
@@ -1,968 +0,0 @@
-import { Morph, TilingLayout, touchInputDevice, morph, Icon, easings } from 'lively.morphic';
```
Comment (merryman) [inline lively.ide/studio/styling.js:1]: Suprised we did not remove that already.

File: lively.ide/value-widgets.js [hunk 2]
```diff
@@ -320,6 +320,7 @@ export class NumberWidget extends Morph {
   }
 
   update (v, fromScrubber = true) {
+    if (this.inactive) return;
```
Comment (merryman) [inline lively.ide/value-widgets.js:323]: Where is this set?

**Review by merryman — COMMENTED** [review]
Looks good overall, I will add some design revisions as soon as you are done and then this is ready.

File: lively.components/inputs.cp.js [hunk 3]
```diff
@@ -43,7 +43,6 @@ const LabeledCheckBox = component({
     axis: 'row',
     wrapSubmorphs: false,
     align: 'top',
-    autoResize: true,
     direction: 'leftToRight',
     resizeSubmorphs: false
   }),
```
[the remark this answers is not in the document]
Comment (linusha) [reply lively.components/inputs.cp.js:48]: Good call, I removed them in all `TilingLayout` definitions.

File: lively.ide/value-widgets.js (hunk 2 above)
Comment (linusha) [reply lively.ide/value-widgets.js:323]: Good catch, that was a half-baked idea that was not necessary in the end. Removed!

File: lively.ide/studio/controls/shape.cp.js [hunk 4]
```diff
@@ -110,10 +117,144 @@ export class ShapeControlModel extends ViewModel {
...
+    widthModeSelector.enable();
+    heightModeSelector.enable();
+
+    xInput.enable()
+    yInput.enable()
+
+
+    if (targetIsTiling && parentIsTiling) {
+      xInput.disable()
+      yInput.disable()
```
[the remark this answers is not in the document]
Comment (linusha) [reply lively.ide/studio/controls/shape.cp.js:159]: Makes sense to me, changed.

File: lively.ide/studio/controls/layout.cp.js [hunk 5]
```diff
@@ -182,6 +185,9 @@ export class AutoLayoutControlModel extends PropertySectionModel {
       layoutableSubmorphs.forEach(m => m.position = m.position);
     }
     this.popup = false;
+    
+    const propertiesPanel = this.view.ownerChain().find(m => m.isPropertiesPanel)
+    if (propertiesPanel) propertiesPanel.viewModel.ui.shapeControl.refreshFromTarget();
```
[the remark this answers is not in the document]
Comment (linusha) [reply lively.ide/studio/controls/layout.cp.js:190]: Done.

File: lively.ide/studio/controls/layout.cp.js [hunk 6]
```diff
@@ -168,6 +168,9 @@ export class AutoLayoutControlModel extends PropertySectionModel {
     const layout = this.targetMorph && this.targetMorph.layout;
     if (!layout || layout.name() !== 'Tiling') { this.targetMorph.layout = new TilingLayout(); }
     this.update();
+    
+    const propertiesPanel = this.view.ownerChain().find(m => m.isPropertiesPanel)
+    if (propertiesPanel) propertiesPanel.viewModel.ui.shapeControl.refreshFromTarget();
```
[the remark this answers is not in the document]
Comment (linusha) [reply lively.ide/studio/controls/layout.cp.js:173]: Done

**Review by merryman — APPROVED** [review]

**Pull request merged**